### PR TITLE
port: enable LG syntax in OAuthInput

### LINKED ISSUE #3306: port: Enable LG syntax in OAuthInput Text and Title (#5186)
The changes in [Enable LG syntax in OAuthInput Text and Title (#5186)](https://github.com/microsoft/botbuilder-dotnet/pull/5186) may need to be ported to maintain parity with `microsoft/botbuilder-dotnet`.

<blockquote>
Fixes https://github.com/microsoft/botbuilder-dotnet/issues/4960
</blockquote>

Please review and, if necessary, port the changes.

### PR DESCRIPTION
**Pull request by chon219**
Fixes #3306 

## Description

Enabled LG syntax in `OAuthInput`'s title and text.
Fixed blocking bugs as well.

## Specific Changes

  - Refactored `OAuthPrompt` to expose `sendOAuthCard()`
  - Updated `OAuthInput` to enable LG syntax
  - Fixed a blocking bug in Expression.parse() to trim starting `=` of expressions
  - Fixed a blocking bug in `Templates` to recognize multi line text correctly

## Testing

Ported tests as well.

### CODE REVIEW

File: libraries/botbuilder-dialogs/src/prompts/oauthPrompt.ts [hunk 1]
```diff
@@ -229,9 +228,9 @@ export class OAuthPrompt extends Dialog {
         const isMessage: boolean = dc.context.activity.type === ActivityTypes.Message;
         const isTimeoutActivityType: boolean =
             isMessage ||
-            this.isTokenResponseEvent(dc.context) ||
-            this.isTeamsVerificationInvoke(dc.context) ||
-            this.isTokenExchangeRequestInvoke(dc.context);
+            OAuthPrompt.isTokenResponseEvent(dc.context) ||
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs/src/prompts/oauthPrompt.ts:231]: Since you're here could you switch these to `??`

**Review by joshgummersall — COMMENTED** [review]
One small ask, otherwise looks good.

File: libraries/botbuilder-dialogs/src/prompts/oauthPrompt.ts (hunk 1 above)
Comment (chon219) [reply libraries/botbuilder-dialogs/src/prompts/oauthPrompt.ts:231]: They are boolean values, so I believe we should use `||`.

Comment (joshgummersall) [reply libraries/botbuilder-dialogs/src/prompts/oauthPrompt.ts:231]: Ah - of course, oversight on my part. Thanks!

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**